grammar/it/src/rules.rs

```diff
@@ -0,0 +1,513 @@
+                          IntegerValue::new(value)
+                      }
+    );
+    b.rule_1_terminal("number (20..90)",
+                      b.reg(r#"(venti|trenta|(?:(?:quar|cinqu|sess|sett|ott|nov)anta))"#)?,
+                      |text_match| {
+                          let value = match text_match.group(1).as_ref() {
+                              "venti" => 20,
+                              "trenta" => 30,
+                              "quaranta" => 40,
+                              "cinquanta" => 50,
+                              "sessanta" => 60,
+                              "settanta" => 70,
+                              "ottanta" => 80,
+                              "novanta" => 90,
+                              _ => return Err(RuleError::Invalid.into()),
+                          };
+                          IntegerValue::new(value)
+                      });
+    b.rule_2("number (21..29 31..39 41..49 51..59 61..69 71..79 81..89 91..99)",
+             b.reg(r#"(venti?|trenta?|(?:(?:quar|cinqu|sess|sett|ott|nov)anta?))"#)?,
+             integer_check_by_range!(1, 9),
+             |text_match, b| {
+                 let value = match text_match.group(1).as_ref() {
+                     "venti" => 20,
+                     "trenta" => 30,
+                     "quaranta" => 40,
+                     "cinquanta" => 50,
+                     "sessanta" => 60,
+                     "settanta" => 70,
+                     "ottanta" => 80,
+                     "novanta" => 90,
+                     "vent" => 20,
+                     "trent" => 30,
+                     "quarant" => 40,
+                     "cinquant" => 50,
+                     "sessant" => 60,
+                     "settant" => 70,
+                     "ottant" => 80,
+                     "novant" => 90,
+                     _ => return Err(RuleError::Invalid.into())
+                 };
+                 IntegerValue::new(value + b.value().value)
+             });
+    b.rule_1_terminal("number 100..1000",
+                      b.reg(r#"(cento?|duecento|trecento|quattrocento|cinquecento|seicento|settecento|ottocento|novecento|mil)"#)?,
+                      |text_match| {
+                          let value = match text_match.group(1).as_ref() {
+                              "cent" => 100,
+                              "cento" => 100,
+                              "duecento" => 200,
+                              "trecento" => 300,
+                              "quattrocento" => 400,
+                              "cinquecento" => 500,
+                              "seicento" => 600,
+                              "settecento" => 700,
+                              "ottocento" => 800,
+                              "novecento" => 900,
+                              "mil" => 1000,
+                              _ => return Err(RuleError::Invalid.into())
+                          };
+                          IntegerValue::new(value)
```

Comment:
You should use the `new_with_grain`. Actually it is grain 2

Reply:
OK, will change